docs/calendar.md

```diff
@@ -347,6 +347,38 @@ Temporal.Calendar.from('iso8601').fields(['month', 'day']);
 ```
 <!-- prettier-ignore-end -->
 
+### calendar.**mergeFields**(_fields_: object, _extraFields_: object) : object
+
+**Parameters:**
+
+- `fields` (object): An object with properties representing calendar units.
+- `extraFields` (object): Another object with properties representing calendar units.
+
+**Returns:** an object with properties from both `fields` and `extraFields`.
+
+This method does not need to be called directly except in specialized code.
+It is called indirectly when using the `with()` methods of `Temporal.PlainDateTime`, `Temporal.PlainDate`, `Temporal.PlainMonthDay`, `Temporal.PlainYearMonth`, and `Temporal.ZonedDateTime`.
+
+Custom calendars should override this method if they allow a calendar unit to be specified in more than one way.
+(For example, the Gregorian calendar allows years to be specified either by a `year` property or a combination of `era` and `eraYear`.)
+
+When subclassing `Temporal.Calendar`, this method doesn't need to be overridden, unless your calendar allows this with other properties than the built-in calendars do.
+The default implementation copies all properties from `extraFields` onto `fields`, taking into account that months may be specified either by `month` or `monthCode` properties, and any other special cases required by built-in calendars.
```

Comment:
As far as I can tell, for all potential usages, fields will have already been created by `ToTemporal____Fields` as the validated new-fields-bag; since those will always return new objects, I think this is explicitly intended but agree it should be explicitly stated and mutation (allowed, expected).